polkadot/primitives/src/v6/mod.rs

```diff
@@ -1328,6 +1363,11 @@ pub enum ValidDisputeStatementKind {
 	/// An approval vote from the approval checking phase.
 	#[codec(index = 3)]
 	ApprovalChecking,
+	/// An approval vote from the new version.
+	/// TODO: Fixme this probably means we can't create this version
+	/// untill all nodes have been updated to support it.
+	#[codec(index = 4)]
+	ApprovalCheckingMultipleCandidates(Vec<CandidateHash>),
```

Comment:
hmm, this is tricky as it's shared between runtime and node-side. once runtime accepts this, we can't prevent other nodes from creating this new version

Reply:
This is how I envisioned we are going to prevent nodes and runtime to use the new type until all the nodes have upgraded. 

https://github.com/paritytech/polkadot-sdk/pull/1191/files#diff-7a698255fbdeac52d6ebd7f6475b8c6e168e57036c0ea37b5439c44abe8feb19R551

What am I missing here, why wouldn't that work ? This type won't be used until we decide to enable approval-coalescing through a runtime upgrade.

Reply:
It's important to remember that anyone can produce a block, including malicious actors. That means once we enable something to be accepted in the runtime, we have to assume someone will (ab)use this. That means we can't rely on filtering on node side only.

Reply:
I guess one option would be to not accept it in the runtime as a valid signature for now and just added it as an enum variant here: https://github.com/paritytech/polkadot-sdk/pull/1191/files#diff-3a256f82ec7286fe77586bafb94fba09556dedd55a0d55cbb1fc249a4bd48814R1281.

The path then for enabling this in production networks would be:
1. Release a node version with the new enum value.
2. Once all nodes updated to 1) release the runtime where the new value is accepted in the runtime.
3. Activate approval-coalescing with the runtime call and nodes would start producing ApprovalCheckingMultipleCandidates.

What do you think ?

Reply:
Yes, a solution to this is to make sure the new runtime is only enacted after supermajority or validators have upgrade their clients to the latest version. This requires some communication and coordination among node runners and the fellowship I guess.

Reply:
That being said, it might be more tricky with polkadot's runtime being applied later than kusama, while node version being rolled out roughtly at the same time.

Reply:
What do you think about this approach: https://github.com/paritytech/polkadot-sdk/pull/1191/commits/569ebb7ee5a9110d347fb47ca2336468d47de3b0 

In this way the runtime won't accept the new enum variant until we are ready to enable it.

The benefit we can update both nodes at runtime at their own pace without the need to coordinate and once we confirm all nodes updated, we can enable this PR by calling set_approval_voting_params.

I think this should work, unless I'm missing something else.